no logging for errors in enb

```sh
yeti-or$ BBIN='...' BBOUT='...' enb make
Error: Cannot find module 'enb-postcss/techs/enb-postcss'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468
...
yeti-or$ mark build
yeti-or$ mark build
touch declaration
prepare config for enb in .../.enb/make.js
clean old data
exec enb
yeti-or$
```

--- a/src/lib/cli/commands/build.js
+++ b/src/lib/cli/commands/build.js
@@ -13,8 +13,5 @@ module.exports = function() {
       .short('i')
       .def('.')
       .end()
-    .act(function(opts) {
-      const {INPUT} = opts;
-      INPUT && build(process.cwd(), INPUT);
-    });
+    .act(opts => build(opts.INPUT));
 };

--- a/src/scripts/build.js
+++ b/src/scripts/build.js
@@ -15,7 +15,8 @@ const decl = [
   {name: 'lang-switcher'}
 ];
 
-module.exports = function (CWD, IN) {
+module.exports = function (IN) {
+  const CWD = process.cwd();
   const INPUT = join(CWD, IN);
   const userConfig = require(join(INPUT, 'config'));
   const OUTPUT = join(CWD, userConfig.output);
@@ -34,7 +35,7 @@ module.exports = function (CWD, IN) {
   }).forEach(layout => layouts.push({name: layout}));
 
   decl.push({name: 'page', mods: [{ name: 'layout', vals: layouts }]});
-  fs.outputFileSync(join(BUNDLE, 'index.bemdecl.js'), `exports.blocks=${JSON.stringify(decl)}`);
+  fs.outputFileSync(join(BUNDLE, 'index.bemdecl.js'), `exports.blocks = ${JSON.stringify(decl, null, 4)}`);
 
   log.verbose('prepare config for enb in', ENB);
   fs.copySync(join(__dirname, 'make.js'), ENB);

--- a/src/scripts/build.js
+++ b/src/scripts/build.js
@@ -2,8 +2,10 @@
 
 const fs = require('bluebird').promisifyAll(require('fs-extra'));
 const {join} = require('path');
-const {exec} = require('child_process');
 const successSymbol = require('log-symbols').success;
+
+const {make} = require('enb');
+const getEnbConfig = require('./make.js');
 
 const grabMd = require('./grab');
 const generateStatic = require('./generate');
@@ -20,8 +22,10 @@ module.exports = function (IN) {
   const INPUT = join(CWD, IN);
   const userConfig = require(join(INPUT, 'config'));
   const OUTPUT = join(CWD, userConfig.output);
+
   const ENB_DIR = join(CWD, '.enb');
-  const ENB = join(ENB_DIR, 'make.js');
+  fs.ensureDirSync(ENB_DIR);
+
   const TMP = join(CWD, '.mark');
   const BUNDLE = join(TMP, 'index');
   const log = require('../lib/log')(userConfig.debug);
@@ -37,50 +41,42 @@ module.exports = function (IN) {
   decl.push({name: 'page', mods: [{ name: 'layout', vals: layouts }]});
   fs.outputFileSync(join(BUNDLE, 'index.bemdecl.js'), `exports.blocks = ${JSON.stringify(decl, null, 4)}`);
 
-  log.verbose('prepare config for enb in', ENB);
-  fs.copySync(join(__dirname, 'make.js'), ENB);
-
   log.verbose('clean old data');
   fs.removeSync(OUTPUT);
   fs.removeSync(join(TMP, 'data.json'));
 
   log.verbose('exec enb');
-  exec(`BBIN=${INPUT} BBOUT=${OUTPUT} enb make`, (error, stdout, stderr) => {
-    if(error) {
-      throw new Error(error);
-      console.error(error);
-      console.log('\n');
-    };
 
-    if(userConfig.debug) {
-      console.log(stdout);
-      console.log('\n');
-    }
+  make({config: getEnbConfig(userConfig)})
+    .then(() => {
+      log.verbose('init grabbing');
+      grabMd(userConfig, CWD, INPUT, OUTPUT, log);
 
-    log.verbose('init grabbing');
-    grabMd(userConfig, CWD, INPUT, OUTPUT, log);
+      log.verbose('init generation');
+      generateStatic(userConfig, CWD, INPUT, OUTPUT, log);
 
-    log.verbose('init generation');
-    generateStatic(userConfig, CWD, INPUT, OUTPUT, log);
+      log.verbose('ensure nojekyll file');
+      fs.ensureFileSync(join(OUTPUT, '.nojekyll'));
+    })
+    .then(() => {
+      log.verbose('move assets');
+      return Promise.all([
+        fs.moveAsync(
+          join(BUNDLE, 'index.min.js'),
+          join(OUTPUT, 'js', 'scripts.min.js')
+        ),
+        fs.moveAsync(
+          join(BUNDLE, 'index.min.css'),
+          join(OUTPUT, 'css', 'styles.min.css')
+        )
+      ]);
+    })
+    .then(() => {
+        log.verbose('clean enb temp');
+        fs.removeSync(ENB_DIR);
+        fs.removeSync(TMP);
 
-    log.verbose('ensure nojekyll file');
-    fs.ensureFileSync(join(OUTPUT, '.nojekyll'));
-
-    log.verbose('move assets');
-    Promise.all([
-      fs.moveAsync(
-        join(BUNDLE, 'index.min.js'),
-        join(OUTPUT, 'js', 'scripts.min.js')
-      ),
-      fs.moveAsync(
-        join(BUNDLE, 'index.min.css'),
-        join(OUTPUT, 'css', 'styles.min.css')
-      )
-    ]).then(() => {
-      log.verbose('clean enb temp');
-      fs.removeSync(ENB_DIR);
-
-      log.info(`[${successSymbol}]`, 'all done');
-    });
-  });
+        log.info(`[${successSymbol}]`, 'all done');
+    })
+    .fail(err => log.error(err))
 }

--- a/src/scripts/grab.js
+++ b/src/scripts/grab.js
@@ -1,17 +1,19 @@
 'use strict';
 
 const fs = require('bluebird').promisifyAll(require('fs-extra'));
-const {join} = require('path');
+const {join, basename, dirname} = require('path');
 const marked = require('meta-marked');
 const glob = require('glob');
 const posthtml = require('posthtml');
 const infoSymbol = require('log-symbols').info;
+const defPosthtmlPlugins = require('../../').posthtmlPlugins;
 
 const defContentDir = 'content';
 
 module.exports = function (userConfig, CWD, INPUT, OUTPUT, log) {
+
   log.info(`[${infoSymbol}]`, `grab *.md files from ${INPUT}...`);
-  const TMP = join(CWD, '.mark');
+  const TMP = join(process.cwd(), '.mark');
   const dataPath = join(TMP, 'data.json');
   const mdFilesInRoot = glob.sync(join(INPUT, defContentDir, '*.md'));
   const mdFilesWithLayout = glob.sync(join(INPUT, defContentDir, '*', '*.md'));
@@ -21,28 +23,27 @@ module.exports = function (userConfig, CWD, INPUT, OUTPUT, log) {
     log.verbose('compile', file);
     const md = fs.readFileSync(file, 'utf-8');
     const compiled = marked(md);
-    const fileName = file.split('/').reverse()[0];
-    const pageName = fileName.split('.')[0];
-    const layoutTest = file.split('/').reverse()[1];
+    const fileName = basename(file);
+    const layoutTest = basename(dirname(file));
     const layout = layoutTest === defContentDir ? 'root' : layoutTest;
 
     const posthtmlPlugins = (userConfig.layouts &&
                             userConfig.layouts[layout] &&
                             userConfig.layouts[layout].posthtmlPlugins) ||
                             userConfig.posthtmlPlugins ||
-                            require('mad-mark').posthtmlPlugins;
+                            defPosthtmlPlugins;
 
     return {
       fileName: fileName,
-      name: pageName,
+      name: fileName.split('.')[0],
       layout: layout,
       path: file,
-      lang: file.split('.').reverse()[1],
+      lang: fileName.split('.')[1],
       meta: compiled.meta,
       content: posthtml(posthtmlPlugins).process(compiled.html, { sync: true }).tree
     };
   });
 
   log.verbose('collect data to', dataPath);
-  fs.outputFileSync(dataPath, JSON.stringify(results));
+  fs.outputFileSync(dataPath, JSON.stringify(results, null, 4));
 }

--- a/src/scripts/make.js
+++ b/src/scripts/make.js
@@ -1,10 +1,6 @@
 'use strict';
 
 const {join, resolve, basename} = require('path');
-const INPUT = process.env.BBIN;
-const OUTPUT = process.env.BBOUT;
-const userConfig = require(join(INPUT, 'config'));
-const CWD = OUTPUT.replace(userConfig.output, '');
 
 const techs = {
   fileProvider: require('enb/techs/file-provider'),
@@ -15,56 +11,61 @@ const techs = {
   bemhtml: require('enb-bemxjst/techs/bemhtml')
 };
 const enbBemTechs = require('enb-bem-techs');
-const levels = [].concat(
-  require('mad-mark').levels.map(path => (join(CWD, path))),
-  userConfig.themes.map(theme => (`${join(INPUT, 'themes', theme)}`))
-);
 
-module.exports = function(config) {
-  config.nodes(`${join('.mark', '*')}`, nodeConfig => {
-    nodeConfig.addTechs([
-      [enbBemTechs.levels, { levels: levels }],
-      [techs.fileProvider, { target: '?.bemdecl.js' }],
-      [enbBemTechs.deps],
-      [enbBemTechs.files],
-      [techs.postcss, {
-        sourceSuffixes: ['css', 'post.css'],
-        target: '.tmp.css',
-        sourcemap: true,
-        plugins: userConfig.postcssPlugins || []
-      }],
-      [enbBemTechs.depsByTechToBemdecl, {
-        target: '.tmp.bemhtml.bemdecl.js',
-        sourceTech: 'js',
-        destTech: 'bemhtml'
-      }],
-      [enbBemTechs.deps, {
-        target: '.tmp.bemhtml.deps.js',
-        bemdeclFile: '.tmp.bemhtml.bemdecl.js'
-      }],
-      [enbBemTechs.files, {
-        depsFile: '.tmp.bemhtml.deps.js',
-        filesTarget: '?.bemhtml.files',
-        dirsTarget: '?.bemhtml.dirs'
-      }],
-      [techs.bemhtml, {
-        target: '.tmp.browser.bemhtml.js',
-        filesTarget: '?.bemhtml.files',
-        sourceSuffixes: ['bemhtml', 'bemhtml.js']
-      }],
-      [techs.browserJs, {
-        includeYM: true,
-        target: '.tmp.browser.js',
-      }],
-      [techs.fileMerge, {
-        target: '.tmp.js',
-        sources: ['.tmp.browser.js', '.tmp.browser.bemhtml.js']
-      }],
-      [techs.bemhtml, { sourceSuffixes: ['bemhtml', 'bemhtml.js'] }],
-      [techs.fileCopy, { source: '.tmp.js', target: '?.min.js' }],
-      [techs.fileCopy, { source: '.tmp.css', target: '?.min.css' }]
-    ]);
+module.exports = userConfig => {
 
-    nodeConfig.addTargets(['?.bemhtml.js', '?.min.css', '?.min.js']);
-  });
+  const levels = [].concat(
+    require('../../').levels,
+    userConfig.themes.map(theme => (`${join('themes', theme)}`))
+  );
+
+  return function(config) {
+    config.nodes(`${join('.mark', '*')}`, nodeConfig => {
+      nodeConfig.addTechs([
+        [enbBemTechs.levels, { levels: levels }],
+        [techs.fileProvider, { target: '?.bemdecl.js' }],
+        [enbBemTechs.deps],
+        [enbBemTechs.files],
+        [techs.postcss, {
+          sourceSuffixes: ['css', 'post.css'],
+          target: '.tmp.css',
+          sourcemap: true,
+          plugins: userConfig.postcssPlugins || []
+        }],
+        [enbBemTechs.depsByTechToBemdecl, {
+          target: '.tmp.bemhtml.bemdecl.js',
+          sourceTech: 'js',
+          destTech: 'bemhtml'
+        }],
+        [enbBemTechs.deps, {
+          target: '.tmp.bemhtml.deps.js',
+          bemdeclFile: '.tmp.bemhtml.bemdecl.js'
+        }],
+        [enbBemTechs.files, {
+          depsFile: '.tmp.bemhtml.deps.js',
+          filesTarget: '?.bemhtml.files',
+          dirsTarget: '?.bemhtml.dirs'
+        }],
+        [techs.bemhtml, {
+          target: '.tmp.browser.bemhtml.js',
+          filesTarget: '?.bemhtml.files',
+          sourceSuffixes: ['bemhtml', 'bemhtml.js']
+        }],
+        [techs.browserJs, {
+          includeYM: true,
+          target: '.tmp.browser.js',
+        }],
+        [techs.fileMerge, {
+          target: '.tmp.js',
+          sources: ['.tmp.browser.js', '.tmp.browser.bemhtml.js']
+        }],
+        [techs.bemhtml, { sourceSuffixes: ['bemhtml', 'bemhtml.js'] }],
+        [techs.fileCopy, { source: '.tmp.js', target: '?.min.js' }],
+        [techs.fileCopy, { source: '.tmp.css', target: '?.min.css' }]
+      ]);
+
+      nodeConfig.addTargets(['?.bemhtml.js', '?.min.css', '?.min.js']);
+    });
+  };
+
 };

